creator-node/src/routes/nodeSync.js

```diff
@@ -131,7 +133,10 @@ module.exports = function (app) {
 
       // Expose ipfs node's peer ID.
       const ipfs = req.app.get('ipfsAPI')
-      const ipfsIDObj = await getIPFSPeerId(ipfs)
+
+      if (!ipfsIDObj) {
+        ipfsIDObj = await getIPFSPeerId(ipfs)
```

Comment:
so i was thinking more about some of the edge cases here and while this is a fine way to stat, if we want to be even safer there's a few improvements.

1. we can cache this value in the creator-node/src/utils.js file so anyone that call this function will get the benefit as opposed to just this one file
2. we probably want to reset the value in case some config changed eg every ten minutes we refresh. we can do that in code like https://stackoverflow.com/a/62529100/11240376

Reply:
@SidSethi let me know what you think about the above

Reply:
good callouts dheeraj
def agreed on 1).
we'll also need 2) bc i believe this can change on ipfs node restart and maybe some other scenarios